src/Serilog.Sinks.BrowserHttp/Sinks/BrowserHttp/BrowserHttpSink.cs

```diff
@@ -29,136 +29,159 @@
-				}
-				catch (Exception ex)
-				{
-					LogNonFormattableEvent(logEvent, ex);
-					continue;
-				}
-
-				var json = buffer.ToString();
-				if (CheckEventBodySize(json, eventBodyLimitBytes))
-				{
-					payload.WriteLine(json);
-				}
-			}
-
-			return payload.ToString();
-		}
-
-		protected override bool CanInclude(LogEvent evt)
-		{
-			return _controlledSwitch.IsIncluded(evt);
-		}
-
-		static bool CheckEventBodySize(string json, long? eventBodyLimitBytes)
-		{
-			if (eventBodyLimitBytes.HasValue &&
-				Encoding.UTF8.GetByteCount(json) > eventBodyLimitBytes.Value)
-			{
-				SelfLog.WriteLine(
-					"Event JSON representation exceeds the byte size limit of {0} set for this sink and will be dropped; data: {1}",
-					eventBodyLimitBytes, json);
-				return false;
-			}
-
-			return true;
-		}
-
-		static void LogNonFormattableEvent(LogEvent logEvent, Exception ex)
-		{
-			SelfLog.WriteLine(
-				"Event at {0} with message template {1} could not be formatted into JSON for the server and will be dropped: {2}",
-				logEvent.Timestamp.ToString("o"), logEvent.MessageTemplate.Text, ex);
-		}
-	}
+    {
+        public const int DefaultBatchPostingLimit = 1000;
+        public static readonly TimeSpan DefaultPeriod = TimeSpan.FromSeconds(2);
+        public const int DefaultQueueSizeLimit = 100000;
+
+        static readonly TimeSpan RequiredLevelCheckInterval = TimeSpan.FromMinutes(2);
+        static readonly JsonValueFormatter JsonValueFormatter = new JsonValueFormatter();
+
+        readonly string _endpointUrl;
+        readonly long? _eventBodyLimitBytes;
+        readonly HttpClient _httpClient;
+        readonly bool _disposeHttpClient;
+
+        DateTime _nextRequiredLevelCheckUtc = DateTime.UtcNow.Add(RequiredLevelCheckInterval);
+        readonly ControlledLevelSwitch _controlledSwitch;
+
+        public BrowserHttpSink(
+            string endpointUrl,
+            int batchPostingLimit,
+            TimeSpan period,
+            long? eventBodyLimitBytes,
+            LoggingLevelSwitch levelControlSwitch,
+            int queueSizeLimit,
+            HttpMessageHandler messageHandler,
+            IDictionary<string, string> defaultRequestHeaders)
+        : this(messageHandler == null ? new HttpClient() : new HttpClient(messageHandler), endpointUrl, batchPostingLimit, period, eventBodyLimitBytes, levelControlSwitch, queueSizeLimit, defaultRequestHeaders, true) { }
+
+        public BrowserHttpSink(
+            HttpClient httpClient,
+            string endpointUrl,
+            int batchPostingLimit,
+            TimeSpan period,
+            long? eventBodyLimitBytes,
+            LoggingLevelSwitch levelControlSwitch,
+            int queueSizeLimit,
+            IDictionary<string, string> defaultRequestHeaders)
+        : this(httpClient, endpointUrl, batchPostingLimit, period, eventBodyLimitBytes, levelControlSwitch, queueSizeLimit, defaultRequestHeaders, false) { }
+
+        private BrowserHttpSink(
```

Comment:
Since the extension methods already handle juggling parameters, perhaps we make this the only (public) constructor here, and deal with mapping configuration to the various parameters up in the configuration layer?